azurerm/internal/services/customproviders/registration.go

```diff
@@ -0,0 +1,31 @@
+package customproviders
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type Registration struct{}
+
+// Name is the name of this Service
+func (r Registration) Name() string {
+	return "Custom Providers"
+}
+
+// WebsiteCategories returns a list of categories which can be used for the sidebar
+func (r Registration) WebsiteCategories() []string {
+	return []string{
+		"Custom Providers",
+	}
+}
+
+// SupportedDataSources returns the supported Data Sources supported by this Service
+func (r Registration) SupportedDataSources() map[string]*schema.Resource {
+	return map[string]*schema.Resource{}
+}
+
+// SupportedResources returns the supported Resources supported by this Service
+func (r Registration) SupportedResources() map[string]*schema.Resource {
+	return map[string]*schema.Resource{
+		"azurerm_custom_resource_provider": resourceArmCustomResourceProvider(),
```

Comment:
Should this be `azurerm_custom_provider`? not sure what the resource bit means the docs don't seem to include it: https://docs.microsoft.com/en-us/azure/azure-resource-manager/custom-providers/tutorial-custom-providers-create?tabs=azure-cli

and i can't find it in the portal. it would also align better with the next one `azurerm_custom_provider_association`